vm/src/stdlib/time.rs

```diff
@@ -350,17 +350,20 @@ mod time {
         }
 
         fn to_date_time(&self, vm: &VirtualMachine) -> PyResult<NaiveDateTime> {
-            let invalid = || vm.new_value_error("invalid struct_time parameter".to_owned());
+            let invalid_overflow =
+                || vm.new_overflow_error("mktime argument out of range".to_owned());
+            let invalid_value = || vm.new_value_error("invalid struct_time parameter".to_owned());
+
             macro_rules! field {
                 ($field:ident) => {
                     self.$field.clone().try_into_value(vm)?
                 };
             }
             let dt = NaiveDateTime::new(
                 NaiveDate::from_ymd_opt(field!(tm_year), field!(tm_mon), field!(tm_mday))
-                    .ok_or_else(invalid)?,
+                    .ok_or_else(invalid_value)?,
```

Comment:
I think this should raise an `Overflow` error too. The documentation for CPython states:

> If the input value cannot be represented as a valid time, either [OverflowError](https://docs.python.org/3/library/exceptions.html#OverflowError) or [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) will be raised (which depends on whether the invalid value is caught by Python or the underlying C libraries).

The value being caught by Python probably means the conversion from a CPython object into a CPython longobject. I.e, if you pass in a list, this will obviously fail with a `ValueError` because it will be expecting an integer.

The underlying C library is basically equivalent to the `NaiveDate` calls here. I'll try and look into it further on the weekend to see where the value error YunWon mentions happens.

Reply:
That sounds great to me! Do reply in the issue so we can keep as much of the context in one place.